blocks/header/header.js

```diff
@@ -59,31 +71,116 @@ function openOnKeydown(e) {
+        // Toggle the class that controls sub section display
+        subSection.classList.toggle('sub-menu-section-active');
+      }
+    });
+  }
+}
+
+/* ------------------------------ Global Functions ----------------------------------- */
+
+/**
+ * A utility function to create divs with specified ID and classes
+ * @param {string} id The id to associate with the created div element
+ * @param {[string]} classes an array of strings indicating the classes to add to the div element
+ * @returns {HTMLDivElement}
+ */
+function createDiv(id, classes) {
+  const div = document.createElement('div');
+  div.id = id;
+  classes.forEach((className) => {
+    div.classList.add(className);
+  });
+  return div;
+}
+
 function focusNavSection() {
   document.activeElement.addEventListener('keydown', openOnKeydown);
 }
 
 /**
  * Toggles all nav sections
  * @param {Element} sections The container element
- * @param {string} expanded Whether the element should be expanded or collapsed
+ * @param {boolean} expanded Whether the element should be expanded or collapsed
  */
 function toggleAllNavSections(sections, expanded = false) {
-  sections.querySelectorAll('.nav-sections > ul > li').forEach((section) => {
+  sections.querySelectorAll('.nav-sections > ul > div > li').forEach((section) => {
     section.setAttribute('aria-expanded', expanded);
+    section.parentElement.setAttribute('aria-expanded', expanded);
   });
 }
+
+/**
+ * Toggles the fade in animation for the nav sections in mobile mode
+ * Necessary given that they start and end in a display:none state
+ */function triggerMenuFadeIn(expanded) {
+  const navSections = document.querySelector('.nav-sections');
+  if (!isDesktop.matches) {
+    navSections.querySelectorAll(':scope > ul > div > li').forEach((element) => {
+      if (!expanded) {
+        element.classList.add('animate-fade-in');
+      } else {
+        element.classList.remove('animate-fade-in');
+      }
+    });
+  }
+}
+
 /**
  * Toggles the flyout animation of the nav menu in mobile mode
  */
-function toggleNavBackground(expanded) {
-  const headerBackdrop = document.querySelector('.header-backdrop');
+async function toggleNavBackground(expanded) {
+  const headerBackdrop = document.querySelector('.header.backdrop');
   const logo = document.getElementById('brand-logo-big');
   if (headerBackdrop && headerBackdrop.classList.contains('animate-open-backdrop') === expanded) {
     headerBackdrop.classList.toggle('animate-open-backdrop');
     logo.classList.toggle('logo-white');
   }
+
+  // This seems like a total hack but is the only way I can get this animation to work
+  // By triggering it after a brief delay to allow the display:none on the menu to change
+  // eslint-disable-next-line no-promise-executor-return
+  await new Promise((r) => setTimeout(r, animationDelay));
```

Comment:
Not sure if there's a best practice in the Franklin world for this, but the scenario is that the Vonage page animates elements from a display:none state by first transitioning them to a rendered display property and then after an await modifying the property to trigger the desired animation effect (transform, opacity, etc).  I did something similar here but it felt a little hacky.  Open to any suggestions!